src/js/tech/html5.js

```diff
@@ -477,6 +477,11 @@ class Html5 extends Tech {
    * @method setSrc
    */
   setSrc(src) {
+    let loadstartlistener = () => this.currentSource_ = null;
```

Comment:
Also, `loadstartlintener` should probably be moved to a private prototype method so that removing it works correctly on subsequent calls to setSrc.